fastlane_bot/helpers/txhelpers.py

```diff
@@ -117,578 +52,129 @@ def validate_and_submit_transaction(
-            )
+            self.cfg.logger.info("Discarding non-profitable arb transaction")
             return None
 
-    def build_transaction_with_gas(
-        self,
-        routes: List[Dict[str, Any]],
-        src_amt: int,
-        src_address: str,
-        gas_price: int,
-        max_priority_fee: int,
-        nonce: int,
-        access_list: bool = True,
-        test_fake_gas: bool = False,
-        flashloan_struct: List[Dict[str, int or str]] = None,
-    ):
-        """
-        Builds the transaction to be submitted to the blockchain.
-
-        routes: the routes to be used in the transaction
-        src_amt: the amount of the source token to be sent to the transaction
-        gas_price: the gas price to be used in the transaction
-
-        returns: the transaction to be submitted to the blockchain
-        """
-        value = src_amt if (src_address == self.ConfigObj.NATIVE_GAS_TOKEN_ADDRESS and self.ConfigObj.SELF_FUND) else 0
-        transaction = self.build_transaction_generic(
-                self.construct_contract_function,
-                routes,
-                src_amt,
-                src_address,
-                flashloan_struct,
-                gas_price=gas_price,
-                max_priority_fee=max_priority_fee,
-                nonce=nonce,
-                value=value
-        )
-        if transaction is None:
-            return None
-
-        if test_fake_gas:
-            transaction["gas"] = self.ConfigObj.DEFAULT_GAS
-            return transaction
+    def check_and_approve_tokens(self, tokens: List):
+        """
+        This function checks if tokens have been previously approved from the wallet address to the Arbitrage contract.
+        If they are not already approved, it will submit approvals for each token specified in Flashloan tokens.
+        :param tokens: the list of tokens to check/approve
+        """
+
+        for token_address in [token for token in tokens if token != self.cfg.NATIVE_GAS_TOKEN_ADDRESS]:
+            token_contract = self.cfg.w3.eth.contract(address=token_address, abi=ERC20_ABI)
+            allowance = token_contract.caller.allowance(self.wallet_address, self.arb_contract.address)
+            self.cfg.logger.info(f"Remaining allowance for token {token_address} = {allowance}")
+            if allowance == 0:
+                function = token_contract.functions.approve(self.arb_contract.address, MAX_UINT256)
+                tx = self._build_transaction(function=function, value=0)
+                if tx is not None:
+                    raw_tx = self.cfg.w3.eth.account.sign_transaction(tx, self.cfg.ETH_PRIVATE_KEY_BE_CAREFUL).rawTransaction
+                    tx_hash = self.cfg.w3.eth.send_raw_transaction(raw_tx)
+                    self._wait_for_transaction_receipt(tx_hash)
+
+    def _build_transaction(self, function, value):
+        tx_details = {
```

Comment:
Do you know if all chains currently supported & planned support EIP 1559 transactions? I recommend retaining it unless you are confident it's no longer in use

Reply:
All chains currently supported in the bot (Ethereum, Base, Fantom and Mantle) support EIP1559.
Unused code (legacy-transaction support, in this case) should be removed.